pkg/engine/datasource/kafka_datasource/kafka_consumer_group.go

```diff
@@ -123,14 +124,53 @@ func (k *KafkaConsumerGroup) startConsuming(handler sarama.ConsumerGroupHandler)
 		}
 	}()
 
-	// Blocking call
-	err := k.consumerGroup.Consume(k.ctx, []string{k.options.Topic}, handler)
-	if err != nil {
-		k.log.Error("KafkaConsumerGroup.startConsuming",
+	k.wg.Add(1)
+	go func() {
+		defer k.wg.Done()
+
+		// Errors returns a read channel of errors that occurred during the consumer life-cycle.
+		// By default, errors are logged and not returned over this channel.
+		// If you want to implement any custom error handling, set your config's
+		// Consumer.Return.Errors setting to true, and read from this channel.
+		for err := range k.consumerGroup.Errors() {
+			k.log.Error("KafkaConsumerGroup.Consumer",
+				log.String("topic", k.options.Topic),
+				log.String("groupID", k.options.GroupID),
+				log.String("clientID", k.options.ClientID),
+				log.Error(err))
+		}
+	}()
+
+	// From Sarama documents:
+	//
+	// This method should be called inside an infinite loop, when a
+	// server-side rebalance happens, the consumer session will need to be
+	// recreated to get the new claims.
+	for {
+		select {
+		case <-k.ctx.Done():
+			return
+		default:
+		}
+
+		k.log.Info("KafkaConsumerGroup.consumerGroup.Consume has been called",
```

Comment:
I can't help but think we can do something here with an extraction, because we're logging the exact same things, but one is in an Info, and one is an Error (with an error).

Also, we log an Info as well as an Error if we do get an error. Does that make sense?

TLDR: can we extract this into a function and do EITHER a k.log.Info or an k.log.Error?